fcrepo-persistence-ocfl/src/test/java/org/fcrepo/persistence/ocfl/impl/NonRdfSourcePersisterTest.java

```diff
@@ -55,66 +61,171 @@
+
+    private static final String EXTERNAL_URL = "http://example.com/file.txt";
+
+    private static final String EXTERNAL_HANDLING = "proxy";
+
+    private static final Long EXTERNAL_CONTENT_SIZE = 526632l;
 
     private final NonRdfSourcePersister persister = new NonRdfSourcePersister();
 
     @Before
-    public void setUp() {
+    public void setUp() throws Exception {
         when(mapping.getOcflObjectId()).thenReturn("object-id");
-        when(mapping.getParentFedoraResourceId()).thenReturn(parentResourceId);
+        when(mapping.getParentFedoraResourceId()).thenReturn(PARENT_RESOURCE_ID);
+
+        when(session.write(anyString(), any(InputStream.class))).thenReturn(writeOutcome);
+
+        nonRdfSourceOperation = mock(NonRdfSourceOperation.class, withSettings().extraInterfaces(
+                CreateResourceOperation.class));
+        when(nonRdfSourceOperation.getUserPrincipal()).thenReturn(USER_PRINCIPAL);
+        when(nonRdfSourceOperation.getResourceId()).thenReturn(RESOURCE_ID);
+        when(nonRdfSourceOperation.getContentSize()).thenReturn(null);
+
+        when(writeOutcome.getContentSize()).thenReturn(LOCAL_CONTENT_SIZE);
     }
 
     @Test
-    public void testNonRdf() throws Exception {
-
-        final Node resourceUri = createURI(resourceId);
-        final String inputContent = "this is some example content";
-        final InputStream content = IOUtils.toInputStream(inputContent, "UTF-8");
-
-        //create some test server triples
-        final Stream<Triple> serverTriples = Stream.of(Triple.create(resourceUri,
-                RDF.type.asNode(),
-                NON_RDF_SOURCE.asNode()));
-        final RdfStream serverTriplesStream = new DefaultRdfStream(resourceUri, serverTriples);
-        when(nonRdfSourceOperation.getResourceId()).thenReturn(resourceId);
+    public void testNonRdfNewResource() throws Exception {
+
+        final InputStream content = IOUtils.toInputStream(CONTENT_BODY, "UTF-8");
+
         when(nonRdfSourceOperation.getContentStream()).thenReturn(content);
-        when(nonRdfSourceOperation.getServerManagedProperties()).thenReturn(serverTriplesStream);
+        when(nonRdfSourceOperation.getType()).thenReturn(CREATE);
+        when(((CreateResourceOperation) nonRdfSourceOperation).getInteractionModel())
+                .thenReturn(NON_RDF_SOURCE.toString());
 
-        persister.persist(session, nonRdfSourceOperation, mapping);
+        persister.persist(storageSession, session, nonRdfSourceOperation, mapping);
 
         // verify user content
         verify(session).write(eq("child"), userContentCaptor.capture());
         final InputStream userContent = userContentCaptor.getValue();
-        assertEquals(inputContent, IOUtils.toString(userContent, StandardCharsets.UTF_8));
+        assertEquals(CONTENT_BODY, IOUtils.toString(userContent, StandardCharsets.UTF_8));
 
         //verify server triples
```

Comment:
Should probably change this to `// Verify headers`

Reply:
updated. They are sort of server managed triples, but not stored as triples. The comment was lazy copy pasta